backend/app/api/admin_routes/evaluation/evaluation_task.py

```diff
@@ -0,0 +1,172 @@
+from typing import Optional, List
+
+from fastapi import APIRouter, status, HTTPException, Depends
+from fastapi_pagination import Params, Page
+from sqlalchemy import func
+from sqlmodel import select, case, desc
+
+from app.api.admin_routes.evaluation.models import CreateEvaluationTask, EvaluationTaskSummary
+from app.file_storage import default_file_storage
+from app.models import EvaluationTask, EvaluationItem, EvaluationStatus, EvaluationDataset, EvaluationDatasetItem
+from app.api.deps import SessionDep, CurrentSuperuserDep
+
+import pandas as pd
+from fastapi_pagination.ext.sqlmodel import paginate
+
+from app.tasks.evaluate import add_evaluation_task
+from app.types import MimeTypes
+
+router = APIRouter()
+
+
+@router.post("/admin/evaluation/task")
+def create_evaluation_task(
+    evaluation_task: CreateEvaluationTask,
+    session: SessionDep,
+    user: CurrentSuperuserDep
+) -> Optional[EvaluationTask]:
+    """
+    Create an evaluation task from the evaluation dataset.
+
+    Args:
+        evaluation_task.name: The name of the evaluation task.
+        evaluation_task.evaluation_dataset_id: The ID of the uploaded evaluation dataset.
+        evaluation_task.chat_engine: The chat engine to evaluate the queries against. Default is "default".
+        evaluation_task.run_size: The number of queries to evaluate. Default is None, which means all queries in the CSV file.
+
+    Returns:
+        True if the evaluation task is created successfully.
+    """
+
+    name = evaluation_task.name
+    evaluation_dataset_id = evaluation_task.evaluation_dataset_id
+    chat_engine = evaluation_task.chat_engine
+    run_size = evaluation_task.run_size
+
+    dataset = session.get(EvaluationDataset, evaluation_dataset_id)
+
+    # dataset exists checker
+    if not dataset or dataset.user_id != user.id:
+        raise HTTPException(
+            status_code=status.HTTP_404_NOT_FOUND,
+            detail="Evaluation dataset not found",
+        )
+
+    # create evaluation items
+    # caveat: Do the deep copy on purpose to avoid the side effect of the original dataset modification
+    evaluation_items = [EvaluationItem(
+        status=EvaluationStatus.NOT_START,
+        chat_engine=chat_engine,
+        query=item.query,
+        reference=item.reference,
+        retrieved_contexts=item.retrieved_contexts,
+        extra=item.extra,
+    ) for item in dataset.evaluation_data_list]
+
+    evaluation_task = EvaluationTask(
+        name=name,
+        user_id=user.id,
+        evaluation_items=evaluation_items,
+        dataset_id=evaluation_dataset_id,
+    )
+
+    session.add(evaluation_task)
+    session.commit()
+
+    add_evaluation_task.delay(evaluation_task.id)
+
+    return evaluation_task
+
+
+@router.get("/admin/evaluation/task-summary/{evaluation_task_id}")
```

Comment:
```suggestion
@router.get("/admin/evaluation/tasks/{evaluation_task_id}/summary")
```

Reply:
Changed, but for only one task, it will be the `/admin/evaluation/task/{evaluation_task_id}/summary`

Reply:
Whether to use singular or plural forms is a controversial topic, but in most APIs I’ve encountered, it’s common practice to consistently use the plural form. Additionally, since our existing code already follows this name convention, new code should also follows the same convention.

- https://www.reddit.com/r/AskProgramming/comments/15rxwfd/singular_or_plural_nouns/
- https://stackoverflow.com/questions/6845772/should-i-use-singular-or-plural-name-convention-for-rest-resources

GitHub API exmaple:
- https://docs.github.com/en/rest/checks/runs?apiVersion=2022-11-28#get-a-check-run

The existing API:
https://github.com/pingcap/autoflow/blob/1a369a65a1aa7a11bda22d92d3643533980419fc/backend/app/api/admin_routes/chat_engine.py#L26

Reply:
Got it, I'm persuaded.